test/parallel/test-http2-server-socket-destroy.js

```diff
@@ -58,5 +59,8 @@ server.on('listening', common.mustCall(() => {
 
   req.on('aborted', common.mustCall());
   req.resume();
-  req.on('end', common.mustCall());
+
+  try {
+    await once(req, 'end');
```

Comment:
Any reason for using `once()`? Wasn't `req.on('end', common.mustCall());` sufficient?

Reply:
https://github.com/szmarczak/node/blob/44d5a14e79fce5956973bc468c64455a3acf11bb/test/parallel/test-http2-server-socket-destroy.js#L52-L54

Reply:
> Wasn't req.on('end', common.mustCall()); sufficient?

No in case of an error.

Reply:
Doesn't this invalidate that specific condition? I mean it now throws also for `err.code === 'ECONNRESET'` if that is wanted why keeping the check in place?

Reply:
Because we want either `end` or `error` :) If `end` is received then this check passes, if `error` gets emitted then the promise reject and there's an empty catch.

Reply:
Ah ok, I understand. Wouldn't it be more readable/easier to understand to remove the listener the `common.mustCall()` listener `req.on('end', common.mustCall())` and replace it with `common.mustNotCall()` from the listener of the `'error'` event?

Reply:
Nevermind `common.mustCall()` once created must be called.